release/models/network-instance/openconfig-network-instance.yang

```diff
@@ -478,7 +478,10 @@ module openconfig-network-instance {
               A connection connecting a source table to a destination
               table implies that routes that match the policy specified
               for the connection are available for the destination
-              protocol to advertise, or match within its policies.";
+              protocol to advertise, or match within its policies.
+              Implemtations of this connection should ensure that the leaked 
+              routes mustn't participate in the route selection process in
+              the destination table";
```

Comment:
Is this what is really intended?  IMO, isn't the purpose of 'redistributing' routes from one protocol to another is to use them?   (subject to policies defined)

Reply:
Well, if you meant, route-redistribution from one protocol to another is to actually make the redistributed route available for advertisements in the destination protocol to peers rather than influencing the forwarding decisions in the local router, then yes thats what I intend to explicitly state here.

We have 2 kinds of implementations that are impacted by Table connections. One kind where there is a separate table for each routing-protocol. Then there is another setup which has a single table for all routing-protocols and the redistributed routes also end up in the same table. For this latter implementation, I guess it is important to explicitly define the expectations of the feature. If the redistributed route ends up being active, the concerns are that we may end up in a race condition.

Reply:
Oh, I think I see you're saying the implementation should not insert a RIB or FIB (AFT) entry for redistributed routes?   Is this a more clear way to state the concern?   [This official OC documentation](https://github.com/openconfig/public/blob/0426774f2d7af0dc0a2b913893c8d45644176d65/doc/network_instance_redistribution.md#rib-modeling-approach-used-in-openconfig) includes guidance on multi vs. single RIB.  We could add a reference to this documentation into the yang and elaborate there?

Reply:
The intent of change is to clearly state that redistributed route MUST NOT take part is active/installed/forwarding route selection. Otherwise we will create cyclic dependency and potentially forwarding loops and/or blackholes.
The unspoken rule is that only active/installed/forwarding route could be source of redistrion. 
This has nothing to do with implementation differences.
Jeff Haas point that term "table-connection" could be understand as inserting redistributed toute to destination's protocol table on which best/active rote/path selection is running. What must NOT be a case. Here is my mental model:
```
protocol's                                                        protocol's
    peer                protocol's peer                               peer   
                                                                             
      |                        |                                        ^    
      |                        |                                        |    
+-----+------------------------+----------------------------------------+--+ 
|+----+----------------+       |      +---------------------------------+-+| 
||+---+    Src Protocol|       |      |           Dst Protocol          | || 
|||                    |       |      |                                 | || 
|||                    |       |      |                                 | || 
|||                    |       |      |  #===========================#  | || 
|||                    |       |      |  |                           |  | || 
|||                    |       +------+->|            RIB            |  | || 
|||  #===============# |              |  |                           |  | || 
|||  |               | |              |  #===========================#  | || 
||+->|      RIB      |-+-------+      |    ^       |   |                | || 
||   |               | |       |      |    |       |   |active          | || 
||   #===============# |       |      |            |   +routes +        | || 
||           |     ^   |              |   flag     |    only*  |        | || 
||           |     |   |    active    |  active    |           v        | || 
||        +--+     |   |    routes    |  routes    |    +-------------+ | || 
||        |        |   |     only     |            |    |    route    | | || 
||        v        |   |              |    |       |    |advertisement|-+ || 
||+---------------+|   |       |      |    |       v    +-------------+   || 
||| protol's best ||   |       |      |    | +----------+      ^          || 
|||route selection||   |       |      |    | | protol's |      |          || 
||+---------------+|   |       |      |    | |   best   |      |          || 
||        |        |   |       |      |    | |  route   |    +-+          || 
||        |        |   |       |      |    | |selection |    |            || 
||        |        |   |       |      |    | +----------+    |            || 
||        |        |   |       v      |    |       |         |            || 
||        |        |   |+------------+|    |       |  /=============\     || 
||        |            ||   table    ||    |       |  |redistributed|     || 
||        |      flag  ||connections |+----+-------+->|     RIB     |     || 
||    +---+     active ||incl. policy||    |       |  |             |     || 
||    |         routes |+------------+|    |       |  \=============/     || 
||    |                |              |    |       |                      || 
||    |            |   |              |    |       |                      || 
||    |            |   |              |    |       |                      || 
||    |            |   |              |    |       |                      || 
|+----+------------+---+              +----+-------+----------------------+| 
|     |            |                       |       |                       | 
|     v            |                       |       v                       | 
|+-------------------------------------------------------------------+     | 
||                      network-instance active                      |     | 
||                          route selection                          |     | 
|+-------------------------------------------------------------------+     | 
|                                  |                                       | 
|                                  v                                       | 
|#===================================================================#     | 
||                                                                   |     | 
||                                FIB                                |     | 
||                                                                   |     | 
|#===================================================================#     | 
|                             Network instance                             | 
+--------------------------------------------------------------------------+ 
                                                                             
* for BGP as destination protocol, the "best-external" and "best-inactive"   
could be included CONDITIONALLY.                                             
The condition being prefix do not exists in virtual "redistributed RIB"      
                                                                             
```
Note, "redistributed routes" are by definition allways baes and active.